internal/ethapi/receipt_proof_test.go

```diff
@@ -0,0 +1,71 @@
+package ethapi
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/XinFinOrg/XDC-Subnet/common/hexutil"
+	"github.com/XinFinOrg/XDC-Subnet/core/types"
+	"github.com/XinFinOrg/XDC-Subnet/rlp"
+	"github.com/XinFinOrg/XDC-Subnet/trie"
+)
+
+// implement interface only for testing verifyProof
+func (n *proofPairList) Has(key []byte) (bool, error) {
+	key_hex := hexutil.Encode(key)
+	for _, k := range n.keys {
+		if k == key_hex {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (n *proofPairList) Get(key []byte) ([]byte, error) {
+	key_hex := hexutil.Encode(key)
+	for i, k := range n.keys {
+		if k == key_hex {
+			b, err := hexutil.Decode(n.values[i])
+			if err != nil {
+				return nil, err
+			}
+			return b, nil
+		}
+	}
+	return nil, fmt.Errorf("key not found")
+}
+
+func TestReceiptProof(t *testing.T) {
```

Comment:
i think it is necessary to call  GetReceiptProof function 

is it right?

Reply:
I don't know how to write test for calling function like GetReceiptProof or other GetXXX. In the codebase I cannot find any test like this.

Reply:
if use that way how do we ensure the getReceiptProof will no error when we changed some code

Reply:
actually I don't know. I can only say that in our codebase (and the ethereum codebase xinfin forked) there is no test for api functions.